docs/adr/0024-upgrade-templating-toolset-of-auth-server-emails.md

```diff
@@ -0,0 +1,261 @@
+
+**Integration & QA**: During development templates will be marked as being part of a release group. This could be in the form of a mapped list of template names or some variable associated with the template file. Each group will have a corresponding environment variable flag. When this release group’s flag is enabled any templates that fall under it will be used when generating an email template in that environment; until a template's corresponding release group is enabled it is not used and the old/current template will continue to be served. Release groups will be initially enabled in staging until QA has had an opportunity to thoroughly test and evaluate each template in the group, after which they can be incrementally enabled in production.
+
+## Pros and Cons of the Templating Options
+
+`fxa-auth-server` currently makes use of Mustache to develop email templates. Mustache is a logicless HTML templating framework and has been proven to be super helpful in developing mail components so far, but due to its logicless nature it becomes tedious to include conditional statements in the mail components. The following section will detail upon the various pros and cons of potential libraries/ frameworks that could be used to replace Mustache as a templating solution.
+
+### Option A - Continue to use Mustache
+
+- Description
+  - Keep the email templates as is in Mustache
+- Pros
+  - The effort to refactor all the templates would be eliminated for now
+- Cons
+  - The logicless nature of Mustache would continue to bother us in the future
+  - If we continue to use Mustache for our templates then our front-end frameworks in FxA would continue to be different. We currently have Backbone in the content-server, Mustache in the auth-server, and React everywhere else.
+
+### Option B - Use React server-side to generate static HTML templates
+
+- Description
+  - Template the auth server emails using [React](https://reactjs.org/) server-side and convert them over to [static HTML templates](https://reactjs.org/docs/react-dom-server.html#rendertostaticmarkup). Other pros and cons of React can be found [in this ADR](https://github.com/mozilla/fxa/blob/main/docs/adr/0010-transition-fxa-from-backbone-to-react.md#option-c---move-to-isomorphic-react), though the type of React we're describing here would be stateless.
+- Pros
+  - The logicless nature of the templates will be eliminated once they're refactored in React
+  - We already use React in `fxa-settings`, `fxa-admin-panel`, `fxa-payments-server`, and `fxa-react`. There’s potential to cross-share components with the new email templates.
+  - React offers us many integration options. E.g there is one with [MJML](https://github.com/wix-incubator/mjml-react)
+  - React has already been audited by our security team.
+- Cons
+  - It could be a overkill to use React, since the library comes with far too many features which may not be needed for templating
```

Comment:
It definitely feels like overkill. it would be good to note that using React is a bigger con than listed here as it drastically increases the knowledge overhead for typically back-end development tasks. It's pretty easy to read a EJS template, and I've never used it. React is also designed for heavily interactive web applications like Facebook, it doesn't scale down to 'simple' very well. 

It'd be good to evaluate what kind of developer overhead this will add to email related tasks, as we've had relatively simple changes in front-end SubPlat work quickly balloon into weeks of developer time. Will 'simple' email tweaks suffer a similar fate?

Reply:
I personally don't believe it would make editing emails more difficult in the future since they'd be simple/stateless components and it should be straightforward to figure out what's going on especially if the developer has worked in React before - but on the other hand, maybe it's weird/unexpected since it's an atypical use  and agreed,

> Personally I prefer simple templates, but the "potential to cross-share components with the new email templates" could outweigh that.

@chenba By "simple," do you mean you like how EJS looks like HTML with some JS logic sprinkled in there? I'm actually unsure of how much reuse we can get if we go the React/Tailwind route - I think maybe a few components, but the reuse bit probably doesn't have much weight.

Reply:
> 
> @chenba By "simple," do you mean you like how EJS looks like HTML with some JS logic sprinkled in there?

Sorry I wasn't clear, yes.

>
> I'm actually unsure of how much reuse we can get if we go the React/Tailwind route - I think maybe a few components, but the reuse bit probably doesn't have much weight.

That's unfortunate.  Without much reuse, the effort of using React doesn't seem to be worthwhile.  (That said, I don't have personal experience with EJS either.)